server/services/offer.service.js

```diff
@@ -270,6 +271,25 @@ export const getOffer = async (offerId, user) => {
   return offer;
 };
 
+const generatePaymentDates = (offer) => {
+  const paymentDates = [];
+
+  const numberOfPaymentsToBeMade =
+    (offer.totalAmountPayable - offer.initialPayment) / offer.periodicPayment;
+
+  const { paymentFrequency } = offer;
```

Comment:
This can be defined after paymentDates so as to get the variables for `numberOfPaymentsToBeMade`.

```suggestion
  const { paymentFrequency, totalAmountPayable, initialPayment, periodicPayment  } = offer;
```